lib/launcher.rb

```diff
@@ -4,27 +4,40 @@
 require 'proxy/signal_handler'
 require 'proxy/log_buffer/trace_decorator'
 require 'thread'
+require 'rack'
+require 'webrick'
 
 module Proxy
   class Launcher
     include ::Proxy::Log
 
     attr_reader :settings
 
-    def initialize(settings = SETTINGS)
+    def initialize(settings = Proxy::SETTINGS)
       @settings = settings
+      if @settings.http_server_type == "puma"
+        begin
+          require 'puma'
+          require 'rack/handler/puma'
+          require 'puma-patch'
+        rescue LoadError
+          logger.warn 'Puma was requested but not installed, falling back to webrick'
+          @settings.http_server_type = "webrick"
+        end
+      end
+      @servers = []
     end
 
     def pid_path
-      settings.daemon_pid
+      @settings.daemon_pid
```

Comment:
There's `attr_reader :settings` so why is this change needed?

Reply:
It's mostly used with `@`, however not big deal - removed.